functions/local_functions.go

```diff
@@ -0,0 +1,64 @@
+package functions
+
+type localFunctionRegistryImpl struct {
+	dialect Dialect
+
+	// substrait function name to local function variants
+	scalarFunctions    map[string][]*LocalScalarFunctionVariant
+	aggregateFunctions map[string][]*LocalAggregateFunctionVariant
+	windowFunctions    map[string][]*LocalWindowFunctionVariant
+
+	// local function name to substrait function variants
+	localScalarFunctions    map[string][]*LocalScalarFunctionVariant
+	localAggregateFunctions map[string][]*LocalAggregateFunctionVariant
+	localWindowFunctions    map[string][]*LocalWindowFunctionVariant
+}
+
+func newLocalFunctionRegistry(dialect Dialect, scalarFunctions map[string][]*LocalScalarFunctionVariant, aggregateFunctions map[string][]*LocalAggregateFunctionVariant, windowFunctions map[string][]*LocalWindowFunctionVariant) LocalFunctionRegistry {
```

Comment:
not really clear why this is a function as opposed to just constructing where it is used above.